### PR TITLE
Support IPv6 connections

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/4243|

## Description

This PR includes all the necessary changes to support IPv6 connections in Wazuh.

## Configuration options

```xml
<remote>
    <ipv6>yes/no</ipv6>
</remote>

<auth>
    <ipv6>yes/no</ipv6>
</auth>
```

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind (memcheck and descriptor leaks check)

- [x] Retrocompatibility with older Wazuh versions
- [x] Added unit tests (for new features)

### CODE REVIEW

**Review by Dwordcito — APPROVED** [review]

File: src/shared/validate_op.c [hunk 1]
```diff
@@ -463,6 +463,10 @@ int OS_IsValidIP(const char *ip_address, os_ip *final_ip)
                 /* number of regex captures */
                 int sub_strings_num = regex_match->d_size.prts_str_alloc_size/sizeof(char*);
```
Comment (Dwordcito) [inline src/shared/validate_op.c:464]: could be const this ?

Comment (cborla) [reply src/shared/validate_op.c]: I agree, done.

**Review by cborla — APPROVED** [review]
LGTM

**Review by Dwordcito — APPROVED** [review]

**Review by cborla — APPROVED** [review]
LGTM!

File: src/config/client-config.c [hunk 2]
```diff
@@ -453,9 +461,12 @@ int Read_Client_Enrollment(XML_NODE node, agent * logr){
             os_free(target_cfg->centralized_group);
             os_strdup(node[j]->content, target_cfg->centralized_group);
         } else if (strcmp(node[j]->element, xml_agent_addr) == 0) {
-            if (OS_IsValidIP(node[j]->content, NULL) != 0) {
+            if (OS_IsValidIP(node[j]->content, NULL) == 1) {
```
Comment (cborla) [inline src/config/client-config.c:464]: This configuration, Does not allow CIDR or Netmask?
OS_IsValidIP return 2 when it find CIDR or netmask.

File: src/shared/enrollment_op.c [hunk 3]
```diff
@@ -540,8 +540,7 @@ static int w_enrollment_concat_src_ip(char *buff, const char* sender_ip, const i
 
     if(sender_ip && !use_src_ip) { // Force an IP
         /* Check if this is strictly an IP address using a regex */
-        if (OS_IsValidIP(sender_ip, NULL))
-        {
+        if (OS_IsValidIP(sender_ip, NULL) == 1) {
```
Comment (cborla) [inline src/shared/enrollment_op.c:543]: Does not allow CIDR or Netmask?
OS_IsValidIP return 2 when it find CIDR or netmask.

File: src/client-agent/start_agent.c [hunk 4]
```diff
@@ -51,66 +51,57 @@ bool connect_server(int server_id, bool verbose)
...
-            }
-        } else {
-            tmp_str++;
-        }
-    } else {
-        tmp_str = agt->server[server_id].rip;
+        // server address comes in {hostname}/{ip} format
+        ip_address = strdup(++tmp_str);
+    }
+    if (!ip_address) {
```
Comment (cborla) [inline src/client-agent/start_agent.c:68]: Could it be an `else`?
```
if (tmp_str) {
       ip_address = strdup(++tmp_str);
} else {
        // server address is either a host or a ip
        ip_address = OS_GetHost(agt->server[server_id].rip, 3);
}
```

Server addres could be `{hostname}/` ?

File: src/shared/validate_op.c [hunk 5]
```diff
@@ -606,8 +600,8 @@ int OS_GetIPv4FromIPv6(const char *ip_address, char *dst_ip, size_t dst_size)
 }
 
 /* Expand IPv6 address */
-int OS_ExpandIPv6(const char *ip_address, int cidr, char *dst_ip, size_t dst_size) {
```
Comment (cborla) [inline src/shared/validate_op.c]: Delete `const ` from `char *ip_address` allows to modify ip address, but we should be sure about if fit expanded ip into reserved space to `ip_address`.

**Review by cborla — DISMISSED** [review]
LGTM!

File: src/config/client-config.c (hunk 2 above)
Comment (TomasTurina) [reply src/config/client-config.c:467]: It should be a single IP, so it is not allowed a subnet.

File: src/shared/enrollment_op.c (hunk 3 above)
Comment (TomasTurina) [reply src/shared/enrollment_op.c:543]: Ditto https://github.com/wazuh/wazuh/pull/11268#discussion_r787187266

File: src/client-agent/start_agent.c (hunk 4 above)
Comment (TomasTurina) [reply src/client-agent/start_agent.c:68]: Yes, it could be anything since it comes from the user configuration input.

**Review by cborla — APPROVED** [review]
LGTM

File: src/shared/validate_op.c (hunk 5 above)
Comment (TomasTurina) [reply src/shared/validate_op.c]: Fixed here https://github.com/wazuh/wazuh/pull/11268/commits/bdb7101d3eee2270e345aa0bef67324dbfc8c689

**Review by cborla — APPROVED** [review]
lgtm!

**Review by cborla — DISMISSED** [review]
LGTM!

**Review by cborla — APPROVED** [review]
LGTM

**Review by vikman90 — APPROVED** [review]

**Pull request merged**